hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/CompressionCodecFactory.java

```diff
@@ -199,7 +201,7 @@ public CompressionCodec getCodec(Path file) {
     if (codecs != null) {
       String filename = file.getName();
       String reversedFilename =
-          new StringBuilder(filename).reverse().toString();
+          toLowerCase(new StringBuilder(filename).reverse().toString());
```

Comment:
Extract the StringUtils.toLowerCase().
Probably call it lowerReversedFilename.

Reply:
Fixed as suggested.